integration_tests/src/main/python/arithmetic_ops_test.py

```diff
@@ -1453,3 +1453,37 @@ def test_func(spark):
     # not we want for this test.
     conf_no_ansi = {"spark.sql.ansi.enabled": "false"}
     assert_gpu_and_cpu_are_equal_collect(test_func, conf = conf_no_ansi)
+
+
+# Test that try_* functions falls back to CPU when TRY mode is used
+@allow_non_gpu('ProjectExec')
+@pytest.mark.parametrize('data_gen', integral_gens, ids=idfn)
+def test_try_add_fallback_to_cpu(data_gen):
+    assert_gpu_fallback_collect(
+        lambda spark: binary_op_df(spark, data_gen).selectExpr(
+            "try_add(a, b) as result"), "Add")
```

Comment:
Seems missing tests for try_multiply, try_reminder, try_averrage ?

Reply:
I didn't add it since they pretty much do the same as try_add, try_divide. Added test for try_sum so that it tests inside HashAggregate. Wanted to cover the ops with minimum number of test. Please let me know if you prefer adding tests for all the cases.

Reply:
Yes, better to have them tested too, this is to detect unexpected changes in the future which may break them.
To make the test code as simple as possible, we can have the method as a parameter of the test.
e.g.

```
@allow_non_gpu('ProjectExec')
@pytest.mark.parametrize('data_gen', numeric_gens, ids=idfn)
@pytest.mark.parametrize('try_method, op_name', [('try_add', 'Add'), ('try_subtract', 'Subtract'), ....], ids=idfn)
def test_try_divide_fallback_to_cpu(data_gen, try_method, op_name):
    assert_gpu_fallback_collect(
        lambda spark: binary_op_df(spark, data_gen).selectExpr(
            "{}(a, b) as result".format(try_method)), op_name)
```